src/lib/utilities.py

```diff
@@ -11,9 +11,38 @@
 #     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #     See the License for the specific language governing permissions and
 #     limitations under the License.
+import os
 from typing import List
 
+import yaml
+
+from lib.context import LoggingContext
+
 
 def chunks(full_list: List, chunk_size: int) -> List[List]:
     chunk_size = max(1, chunk_size)
-    return [full_list[i:i + chunk_size] for i in range(0, len(full_list), chunk_size)]
\ No newline at end of file
+    return [full_list[i:i + chunk_size] for i in range(0, len(full_list), chunk_size)]
+
+
+def read_activation_yaml():
+    activation_file_path = '/code/config/activation/gcp_services.yaml'
+    try:
+        with open(activation_file_path, encoding="utf-8") as activation_file:
+            activation_yaml = yaml.safe_load(activation_file)
+    except Exception:
+        activation_yaml = yaml.safe_load(os.environ.get("ACTIVATION_CONFIG", ""))
+    if not activation_yaml:
+        activation_yaml = {}
+    return activation_yaml
```

Comment:
If you allow return `{}` how `load_activated_feature_sets` will work? maybe throw exception here or in `load_activated_feature_sets`

Reply:
Thanks for catching that!
I forgot to add empty list as default value...